docs/src/main/asciidoc/datasource.adoc

```diff
@@ -3,11 +3,13 @@ This guide is maintained in the main Quarkus repository
 and pull requests should be submitted there:
 https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
 ////
-[id="datasources"]
+[id="data-sources-configure"]
```

Comment:
FWIW we completely ignored one of the major problems of these refactorings: we're going to create dead links by changing anchors all over the place. For example https://quarkus.io/guides/datasource#tldr won't redirect to the correct section anyway.

Not requesting this gets fixed here though. This is probably something that should be discussed more globally.

Reply:
@yrodiere Thank you for this feedback, Yoann.

ID=datasource**_s_**, filename=datasource.adoc, anchors=<<ID,Description>>, crosslinks=xref:datasource.adoc#tldr[Description].

ID should be derivation of a filename, which is not the case here. ( example: datasource.adoc -> ID="datasource"....)
I search the repo for the `tldr ` and it returns 0 results, thus I should not break the link you shared. It is not working since the #tldr section doesnt exist anymore (can create a redirection PR for it).
Also, the link you posted would be written in the .adoc like this xref:datasource.adoc#tldr[TLDR], thus the change of the ID is not playing a part here, since the link is a derivation of the filename.

Searched for `<<datasource/s,` but also with 0 findings. Putting this back just in case I am missing something or I searched poorly :)

Then, as you proposed, it would be good if we would move/delete unrelated content to this "Reference", thus we will be creating even more anchors/links changes. This should be tuned as a last step of this refactor. But I would do this lately, when I will have some clearance about future plans :)

However, it is possible (and I will know this next Thursday) that this file will be taken to RHBQ. If not, I would need to ask you for a cooperation in which we would: 1) Prepare the Datasource reference guide 2) tweak the other 3 related guides.

Reply:
>  Also, the link you posted would be written in the .adoc like this xref:datasource.adoc#tldr[TLDR], thus the change of the ID is not playing a part here, since the link is a derivation of the filename.

I was actually talking about links from other websites, e.g. stackoverflow.

> It is not working since the #tldr section doesnt exist anymore

Right that's what I meant. Since we're changing sections about everywhere, we probably need to talk about how we' may be wreaking havoc on existing resources online (e.g. stackoverflow) that link to our docs and whose links will become (slightly or completely) dead.

But again that's not something we'll fix in the scope of this PR.